programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -100,20 +100,38 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
         &self,
         remaining_accounts: &[AccountInfo],
         amount: u64,
+        reduce_only: bool,
         allow_token_account_closure: bool,
     ) -> Result<()> {
         require_msg!(amount > 0, "deposit amount must be positive");
 
-        let token_index = self.bank.load()?.token_index;
+        let mut bank = self.bank.load_mut()?;
+        let token_index = bank.token_index;
+
+        let (amount_i80f48, token_index) = {
+            // Get the account's position for that token index
+            let account = self.account.load()?;
+            let position = account.token_position(token_index)?;
```

Comment:
immediately after this block we load account/position again. Could just move the mut-loading above this block to save cu?